.github/workflows/ci.yml

```diff
@@ -33,18 +39,28 @@ jobs:
           node-version: "12"
 
       - id: "haskell"
+        name: "(Non-Linux only) Install Haskell"
+        if: "${{ runner.os != 'Linux' }}"
         uses: "haskell/actions/setup@v1"
         with:
           enable-stack: true
           # If upgrading Stack, also upgrade it in the lint job below
-          stack-version: "2.7.1"
+          stack-version: "${{ env.STACK_VERSION }}"
           stack-no-global: true
 
+      - name: "(Linux only) Check Stack version and ensure a (dummy) accessible local repository"
+        if: "${{ runner.os == 'Linux' }}"
+        run: |
+          [ "$(stack --numeric-version)" = "$STACK_VERSION" ]
+          echo 'allow-different-user: true' >>/root/.stack/config.yaml
+          git -C .. init
```

Comment:
I'm not sure this is a good idea. It seems there are parts of the build that expect to be in a Git repository; what are the consequences of that repository being faked out?

The issue seems to be that if the Git version is less than 2.18, GitHub switches to just unpacking a tarball instead of doing a clone. That to me suggests the alternative of installing a more recent Git instead, though it would be more verbose. Here's how I'd do it, in a Debian-based Docker image:


    if dpkg --compare-versions $(git --version | cut -d' ' -f3) lt 2.18; then
      . /etc/os-release
      echo "deb http://deb.debian.org/debian $VERSION_CODENAME-backports main" > /etc/apt/sources.list.d/backports.list
      apt-get update -qq
      apt-get install -qq git/$VERSION_CODENAME-backports
    fi

Reply:
You are right. I was trying to avoid making it too verbose. And sure, I can try something just like you have suggested.

Reply:
Done.